### PR TITLE
Modernize room screen UI with code from live streaming app

### PR DESCRIPTION
**Pull request by timrozum**
Changes:

1. Started using SwiftUI and Combine to modernize UI design and simplify code.
2. Removed the old lobby and room screens and replaced with new screens based on the live streaming app.
3. Now there is both video grid and focus layouts. Right now the grid just displays the local participant and the most active 5 remote participants. The app will automatically switch to focus layout when there is a presentation track.
4. All of the video room connect options should be completely unchanged. 
5. There are fewer abstractions now which makes the code easier to understand. The downside is the code is less testable which we are ok with.

These features got removed and I will create tickets to add them back next sprint:

1. Display recording indicator when the room is being recorded.
6. Display network quality level for each participant.
7. Display video track switch off state in UI.
8. Switch camera button.
9. Stats screen.
10. Use deep link to set room name.
11. Show media capture preview screen before connecting to room.

I will also create a new ticket to remove more old code. I don't think I got it all in this PR.

These ^ should all be relatively simple so I think I can handle them in next sprint.

Also note:

1. Test coverage was disabled because it was causing issues with SwiftUI previews. I think this is really a CocoaPods issue. I think there are other fixes but I think we should just switch to SPM soon. I confirmed Firebase is now available as a package so no dependencies are holding this up.
2. Pin participant is a feature the app used to have that I don't think will come back soon. We would have to rethink how it would make sense in new video layouts.
3. I think soon we will add the ability for the user to manually switch between grid and focus layouts. And if focus layout is selected and there is no presentation track it would just display large video for the dominant speaker. 

For video SDK integration see `LocalParticipantManager`, `RemoteParticipantManager`, and `RoomManager`.

https://user-images.githubusercontent.com/1930363/156422272-0e1a5acc-e8a7-42f6-9183-2f0a564fa761.mov

### CODE REVIEW

File: VideoApp/VideoApp/NewCode/Screens/Room/RoomViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        roomManager.localParticipant.isCameraOn = true
+
+        Task {
+            do {
+                let accessToken = try await accessTokenStore.fetchTwilioAccessToken(roomName: roomName)
+                roomManager.connect(roomName: roomName, accessToken: accessToken)
+            } catch {
+                handleError(error)
+            }
+        }
```
Comment (timrozum) [inline VideoApp/VideoApp/NewCode/Screens/Room/RoomViewModel.swift:77]: First place I used async/await which Apple has ported back to iOS 13. It made these multiple async calls way cleaner.

File: VideoApp/VideoApp/Stores/AppSettings/TwilioEnvironment.swift [hunk 2]
```diff
@@ -16,7 +16,7 @@
 
 import Foundation
 
-enum Environment: String, SettingOptions {
+enum TwilioEnvironment: String, SettingOptions {
```
Comment (timrozum) [inline VideoApp/VideoApp/Stores/AppSettings/TwilioEnvironment.swift:19]: This was conflicting with a SwiftUI symbol and probably better to be more specific anyway.

File: VideoApp/VideoApp/NewCode/Views/GridLayout/GridLayoutViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+//  limitations under the License.
+//
+
+import TwilioVideo
+import Combine
+
+/// Subscribes to room and participant state changes to provide participant state for the UI to display in a grid.
+class GridLayoutViewModel: ObservableObject {
+    @Published var onscreenParticipants: [ParticipantViewModel] = []
+    @Published var offscreenParticipants: [ParticipantViewModel] = []
```
Comment (timrozum) [inline VideoApp/VideoApp/NewCode/Views/GridLayout/GridLayoutViewModel.swift:23]: This offscreen stuff will go away when I add pagination in next ticket.

File: VideoApp/VideoApp/NewCode/Views/FocusLayout/FocusLayoutViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        dominantSpeaker = findDominantSpeaker()
+        dominantSpeaker.isDominantSpeaker = false // No need to distinguish in the UI since only 1 participant is shown
+    }
+    
+    private func findPresenter() -> Presenter? {
+        guard let presenter = roomManager.remoteParticipants.first(where: { $0.presentationTrack != nil }) else {
+            return nil
+        }
+        
+        return Presenter(identity: presenter.identity, presentationTrack: presenter.presentationTrack!)
```
Comment (timrozum) [inline VideoApp/VideoApp/NewCode/Views/FocusLayout/FocusLayoutViewModel.swift:75]: @ceaglest for this video layout that currently displays the dominant speaker and presentation track I think I might need to be setting track priority. I could set the track priority of the presentation track to high. The dominant speaker track priority is currently set to server default. So will both tracks end up being high priority which kind of defeats the purpose of priority? Maybe only the presentation track should be high priority for a layout like this.

**Review by wolfspy — APPROVED** [review]

File: VideoApp/VideoApp/NewCode/Views/FocusLayout/FocusLayoutViewModel.swift (hunk 4 above)
Comment (timrozum) [reply VideoApp/VideoApp/NewCode/Views/FocusLayout/FocusLayoutViewModel.swift:75]: Will talk about this one when he is back in the office.

**Pull request merged**